src/openapi/streaming/connection/connection.js

```diff
@@ -39,11 +39,12 @@ function onTransportFail(error) {
 
     this.transport.setReceivedCallback(this.receiveCallback);
     this.transport.setStateChangedCallback(this.stateChangedCallback);
+    this.transport.setUnauthorizedCallback(this.unauthorizedCallback);
     this.transport.setConnectionSlowCallback(this.connectionSlowCallback);
     this.transport.setErrorCallback(this.errorCallback);
 
     if (this.state === STATE_STARTED) {
-        this.transport.updateQuery(this.authToken, this.contextId);
+        this.transport.updateQuery(this.authToken, this.contextId, this.authExpiry);

```

Comment:
we pass the authExpiry into the transport so it can pass it back when there is an error to tell us which token was used that generated a auth error.
The trouble is that signal-r would require a few internal changes to tell us this, so signal-r doesn't use it. I assume I can make raw websockets work withit.